File: backend/py_bloom_filter/naive_array.py

class NaiveArray:

    def __init__(self):
        self.entries = []
        self.byte_size = 0

    def check(self,possible_entry):
        return possible_entry in self.entries

    def add(self,new_entry):
        if not self.check(new_entry):
            self.entries.append(new_entry)
            self.byte_size += len(new_entry)

    def size(self):
        return self.byte_size

    def __str__(self):
        return "NaiveArray: size " + str(self.byte_size * 8)

File: backend/c_bloom_filter/naive_array.cpp

#include "naive_array.h"
#include <algorithm>

void NaiveArray::add(std::string new_entry) {
	if (!check(new_entry)) {
		this->entries.push_back(new_entry);
		this->byte_size += new_entry.size();
	}
}

bool NaiveArray::check(std::string possibility) {
	auto vec = this->entries;
	return std::find(vec.begin(),vec.end(),possibility) != vec.end();
}

unsigned long long NaiveArray::size() {
	return this->byte_size;
}

